Add a script for porting passes tests to lit

tlively opened the pull request:
And use it to port the very simple untee test.

aheejin commented on scripts/port_test.py:
```diff
@@ -0,0 +1,104 @@
+    parser.add_argument('--git-add', action='store_true',
+                        help='Stage changes')
+    parser.add_argument('tests', nargs='+', help='The test files to port')
+    args = parser.parse_args()
```
It'd be good to add simple `--help` with a usage string

tlively replied:
This is automatically generated by the arg parser module!

aheejin commented on scripts/port_test.py:
```diff
@@ -0,0 +1,104 @@
+    opts = [('--' + p if not p.startswith('O') and p != 'g' else '-' + p)
+            for p in passes]
+
+    run_line = (f';; RUN: foreach %s %t wasm-opt {" ".join(opts)} -S -o -'
```
Just saw we had `foreach` script: https://github.com/WebAssembly/binaryen/commit/8de4349ebdb6ce398ef4939f9f375a7413d99b09

Was this added in part of a PR? Can't see any link to it..

tlively replied:
Yes, it was hidden in https://github.com/WebAssembly/binaryen/pull/3939

aheejin commented on scripts/port_test.py:
```diff
@@ -0,0 +1,104 @@
+    if not args.no_delete:
+        for f in glob.glob(test.replace('.wast', '.*')):
+            os.remove(f)
+            if args.git_add:
+                subprocess.run(['git', 'add', f])
```
Why `git add` the source file?

tlively replied:
This stages the *removal* of the file.

aheejin reviewed (commented):
Nice!

tlively commented:
I fixed the flake8 error and added a new note to be emitted into the ported test identifying it as automatically ported. I realized this actually depends on https://github.com/WebAssembly/binaryen/pull/3962, so I will wait for that to land before re-generating the test here and landing this.

kripken commented on scripts/port_test.py:
```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python3
```
maybe `port_pass_test_to_lit.py`?

kripken commented on scripts/port_test.py:
```diff
@@ -0,0 +1,110 @@
+                subprocess.run(['git', 'add', f])
+
+    if args.git_add:
+        subprocess.run(['git', 'add', dest])
```
I'm surprised to see the git operations here... I'd expect the user to run them manually. But I don't feel strongly, if this is easier sgtm.

tlively replied:
Yep, it's a little weird to do this in the script, so I put it behind a flag. It's a great quality of life feature, though.
